### PR TITLE
Fix mode_t size on OSX.

### PR DESCRIPTION
**Pull request by toots**

### CODE REVIEW

**Comment by Drup** [comment]
Wow, someone uses that library. Thanks! :D 

Do you think we should use [ocaml-posix-types](https://github.com/yallop/ocaml-posix-types) instead ? It exposes `mode_t` as an unsigned integer (and I would assume it does the correct thing on the various platforms.

File: src/lmdb.ml [hunk 1]
```diff
@@ -58,7 +58,13 @@ module Env = struct
...
 
   let create ?max_readers ?map_size ?max_dbs ?(flags=Flags.none) ?(mode=0o755) path =
-    let mode = coerce uint32_t mode_t UInt32.(of_int mode) in
+    let mode =
+      (* Happens on OSX.. *)
+      if (sizeof mode_t) = 2 then
+        coerce uint16_t mode_t UInt16.(of_int mode)
+      else
+        coerce uint32_t mode_t UInt32.(of_int mode)
+    in
```
Comment (yallop) [inline src/lmdb.ml]: `PosixTypes.Mode.of_int` (see [[1]](https://github.com/ocamllabs/ocaml-ctypes/blob/8d795ddd4d54b11a808c0401359f1885eca4b160/src/ctypes/posixTypes.mli#L17), [[2]](https://github.com/ocamllabs/ocaml-ctypes/blob/8d795ddd4d54b11a808c0401359f1885eca4b160/src/ctypes/unsigned.mli#L47-L48)) is the conversion you need here.

Comment (Drup) [inline src/lmdb.ml]: Yes, that's what I think. When I first wrote this, your library wasn't around, but I have no excuses anymore. :)

**Comment by toots** [comment]
Awesome. @Drup I'm testing the library right now, seems to be working fine. Maybe you could also push it to opam as well :-)

**Comment by Drup** [comment]
Huum, Travis' linux version of lmdb seems quite old, which means that CI is failing all the time. :/

I'll merge anyway, but I would really like to have a way around this.

**Pull request merged**